lib/popup.ts

```diff
@@ -142,6 +147,37 @@ export function setPopupToCreateDom(triggerElem: Element, domCreator: IPopupDomC
   return setPopupToFunc(triggerElem, openFunc, options);
 }
 
+/**
+ * Open a popup using as content the element returned by the given func. Note that the `trigger`
+ * option is ignored by this function and that the default of the `attach` option is `body` instead of `null`.
+ *
+ * It allows you to bind the creation of the popup to a menu item as follow:
+ *   menuItem(elem => popupOpen(elem, (ctl) => buildDom(ctl)))
+ *
+ */
+export function popupOpen(reference: Element, domCreator: IPopupDomCreator, options: IPopupOptions): PopupControl {
+  function openFunc(openCtl: IOpenController) {
+    const content = domCreator(openCtl);
+    function dispose() { domDispose(content); ctl.dispose(); }
+    return {content, dispose};
+  }
+  const ctl = PopupControl.create(null) as PopupControl;
+
+  // Set the default for the attach option to `body`. Because otherwise the default 'null' would
+  // causes the popup to be attached to reference.parentNode. This does make little sense when used
+  // as follow `menuItem((elem) => popupOpen(elem, ...` because it would close the popup just after
+  // the click.
+  options = defaultsDeep(options, {attach: 'body'});
+
+  ctl.attachElem(reference, openFunc, {
+    ...options,
```

Comment:
You can simply add the attach option here, no need for `defaultsDeep` (and I think it's clearer to see how options are combined all together):
```
{
   attach: 'body',
   ...options,
   trigger: undefined,
}
```

Reply:
`defaultsDeep` is slightly better at handling an undefined value which it replaces with the default value. Instead the other approach keeps it to undefined.
```
defaultsDeep({a: undefined}, {a: 1}); // {a: 1}
{a: 1, ...{a: undefined}} // {a: undefined}
```